torchgen/model.py

```diff
@@ -1328,6 +1324,8 @@ def is_list_like(self) -> Optional["ListType"]:
         "Stream",
         "SymInt",
         "ConstQuantizerPtr",  # TODO: rename
+        "ScalarVal",  # An actual aten Scalar value, no special handling
+        "BackendDataPtr",
```

Comment:
@bdhirsh is it ok for us to pollute this list with LT specific types?  it is in service of reusing the parser, which is nice

Reply:
If you can't put it in an IValue, I don't want it in here. Should I read this PR and find out what it's trying to do more broadly?

Reply:
Agreed with "can you put it in an IValue" as a good litmus test

Reply:
Any thoughts on adding a generic cpp type that isn't parsed but used as is? Something like
```
- func: func_name(::CustomCppType arg_name) -> Tensor
```
which would emit something like
```
func_name(CustomCppType arg_name)
```

Reply:
I added a proposed solution for this in the last commit. Please check to see if this is an acceptable solution

Reply:
I'd rather keep a `CustomCppType` out of the data model for the same reason (all of the other codegen, and our JIT type system, wouldn't know how to deal with it, since it's all specific to LTC).

Reply:
> all of the other codegen, and our JIT type system, wouldn't know how to deal with it

Aren't there checks for edge cases like this? If its not a case thats dealt with I'm generally seeing that there are "unsupported type" errors that are thrown. This should be perfectly valid as those don't support custom C++ types. 

Plus there are no ops outside of the non-native ones that exercise this path. I've designed it to only make a custom C++ type if prefixed with `::` (open to suggestions about how to annotate this. I went with this as it's a recognized namespace pattern). Otherwise, it will fail to parse the type, same as before.

Otherwise, there isn't a way that I can think of at the moment to parse those types using the existing function schema parser. I'd need to re-introduce a non-native op parser which seems overkill now for just a couple unsupported types.

Reply:
One escape hatch you have is that you can always manually write out the code for the handful of ops that use this new type (how many ops / IR nodes do you plan to use that new data type?)

The codegen today is mainly meant for aten ops. And there is a long term plan of getting general-purpose codegen to work with custom operators, but the idea is that custom operators should work with the rest of the ecosystem. In the same way that today, you can register custom operators to the dispatcher, but there's an expectation that custom ops support conversion to IValues, work with `__torch_dispatch__`, etc.

Reply:
We should allow torchbind classes in schema strings. I don't know how to make this work but iirc the schema parsing code does know how to handle custom types, so I'm OK with this for torchgen parser too. (BTW, torchbind passes the IValue test)

Reply:
I've kept the `DeviceData` as is then as that was the one op that was causing the most issues. Can you please review again to see if there are any other problematic changes?

Reply:
looks ok to me now. thanks for making the change.

Reply:
Sorry for the late response. Looks like `ScalarVal` and `BackendDataPtr` are still in the enum?

Reply:
ah, I must have forgotten to remove them after making my changes. Removing now